sklearn/cluster/_birch.py

```diff
@@ -156,8 +158,11 @@ def __init__(self, *, threshold, branching_factor, is_leaf, n_features):
         # The list of subclusters, centroids and squared norms
         # to manipulate throughout.
         self.subclusters_ = []
-        self.init_centroids_ = np.zeros((branching_factor + 1, n_features))
-        self.init_sq_norm_ = np.zeros((branching_factor + 1))
+        self.dtype = dtype
```

Comment:
I do not think we need to store the dtype after `init_centroids_` and `init_sq_norm_` are initialized.

```suggestion
```